app/src/main/java/com/android/streetworkapp/ui/utils/CustomDialog.kt

```diff
@@ -45,23 +46,34 @@ fun CustomDialog(
           showDialog.value = false
         },
         confirmButton = {
```

Comment:
Sonar raise an issue about the cognitive complexity of this `CustomDialog` composable function - and I agree with it. To reduce cognitive complexity, you could just implement functions that correspond to the action you want to perform. So it's just a matter of copying and pasting your existing lines into functions and using those functions in the composable. 

Here's an example, instead of having
```kotlin
confirmButton = {
if (dialogType.shouldShowButton()) {
          if (dialogType.shouldShowButton(ButtonType.SUBMIT)) {
            TextButton(
                onClick = {
                  onSubmit()
                  Toast.makeText(context, "Submitted", Toast.LENGTH_SHORT).show()
                  if (verbose) Toast.makeText(context, "Submitted", Toast.LENGTH_SHORT).show()
                  showDialog.value = false
                },
                modifier = Modifier.testTag("${tag}DialogSubmitButton")) {
                  Text("Submit", color = ColorPalette.SECONDARY_TEXT_COLOR)
                }
          } else if (dialogType.shouldShowButton(ButtonType.CONFIRM)) {
            TextButton(
                onClick = {
                  onSubmit()
                  if (verbose) Toast.makeText(context, "Confirmed", Toast.LENGTH_SHORT).show()
                  showDialog.value = false
                },
                modifier = Modifier.testTag("${tag}DialogConfirmButton")) {
                  Text("Confirm", color = Color.Red)
                }
          }
        }
```

You can juste take theses lines, put them in a `handleOnConfirm(...)` function and use it in the composable

```kotlin
confirmButton = handleOnConfirm(...)
```

It should pretty quick, what do you think of it?

Reply:
You are totally right, I will define a handler for the buttons.

Reply:
## New implementation

### `HandleConfirmButton`

Refactoring of the if-logic for which button to display (`SUBMIT`, `CONFRIM`).

### `HandleDismissButton`

Refactoring of the if-logic for which button to display (`CANCEL`).

### `DialogButton`

Handles the display of any button.

- ButtonType - the type of the button (used to extract the text values, testTag, and color).
- tag: String - the tag of the dialog (used for testTag).
- onSubmit: () -> Unit - callback of the button.
- showDialog: MutableState<Boolean> - used to hide the dialog after clicking.
- verbose: Boolean - whether the Toast() should be called (refactoring).

### New functions for `enum class ButtonType`

- fun response(): String - The text for click confirmation (Toast).
- fun tag(): String - The tag for testTag and title of the button.
- fun color(): Color - The color of the button.

Reply:
What a detailed answer! Thank you for making the changes. It's great like this.